src/urlschemehandler.cpp

```diff
@@ -8,11 +8,41 @@
 
 #include <kiwix/search_renderer.h>
 #include <kiwix/name_mapper.h>
+#include <zim/search.h>
+#include <zim/entry.h>
+#include <zim/error.h>
+
 
 UrlSchemeHandler::UrlSchemeHandler()
 {
 }
 
+zim::Entry getEntryFromPath(const zim::Archive& archive, const std::string& path)
+{
+  try {
+    return archive.getEntryByPath(path);
+  } catch (zim::EntryNotFound& e) {
+    if (path.empty() || path == "/") {
+      return archive.getMainEntry();
+    }
+  }
+  throw zim::EntryNotFound("Cannot find entry for non empty path");
+}
+
+zim::Entry getFinalEntry(zim::Entry entry)
+{
+  int loopCounter = 42;
+  auto final_entry = entry;
+  while (final_entry.isRedirect() && loopCounter--) {
+    final_entry = final_entry.getRedirectEntry();
+  }
+  // Prevent infinite loops.
+  if (final_entry.isRedirect()) {
+    throw zim::EntryNotFound("bad entry");
+  }
+  return final_entry;
+}
```

Comment:
`zim::Entry::getItem(true)` follows the redirection (until we have a item) and then return the associated item. We could use it directly instead this intermediate function.